cookbooks/aws-parallelcluster-slurm/recipes/finalize_head_node.rb

```diff
@@ -25,7 +25,26 @@
   block do
     require 'chef/mixin/shell_out'
     require 'shellwords'
+    require 'json'
 
+    def check_for_protected_mode(fleet_status_command)
+      begin
+        cluster_state_json = shell_out!("/bin/bash -c #{fleet_status_command}").stdout.strip
+        cluster_state = JSON.load(cluster_state_json)
+      rescue
+        Chef::Log.warn("Unable to read cluster state")
+        return
+      end
+
+      Chef::Log.info("Cluster state is empty") if cluster_state.empty?
+      return if cluster_state.empty?
+
+      raise "Cluster state has been set to PROTECTED during static node provisioning" if cluster_state["status"] == "PROTECTED"
```

Comment:
What about something similar to: `Cluster has been set to PROTECTED mode due to failures detected in static node provisioning`?

Reply:
Fixed.